PetApiTest/ControllerTest/PetControllerTest.cs

```diff
@@ -0,0 +1,198 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Newtonsoft.Json;
+using PetApi.Model;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using Xunit;
+
+namespace PetApiTest.ControllerTest;
+
+public class PetControllerTest
+{
+    [Fact]
+    public async void Should_add_new_pet_to_system_successfully()
+    {
+        //given
+        var application = new WebApplicationFactory<Program>();
+        var httpClient = application.CreateClient();
+        await httpClient.DeleteAsync(requestUri: "/api/deleteAllPets");
+        var pet = new Pet(name: "Kitty", type: "cat", color: "white", price: 1000);
+        var serializeObject = JsonConvert.SerializeObject(pet);
+        var postBody = new StringContent(serializeObject, Encoding.UTF8, mediaType: "application/json");
+        //when
+        var repsonse = await httpClient.PostAsync(requestUri: "/api/addNewPet", postBody);
+        //then
+        repsonse.EnsureSuccessStatusCode();
+        Assert.Equal(HttpStatusCode.OK, repsonse.StatusCode);
+    }
+
+    [Fact]
+    public async void Should_get_all_pets_of_system_successfully()
+    {
+        //given
+        var application = new WebApplicationFactory<Program>();
+        var httpClient = application.CreateClient();
+        await httpClient.DeleteAsync(requestUri: "/api/deleteAllPets");
+        var pet = new Pet(name: "Kitty", type: "cat", color: "white", price: 1000);
+        var serializeObject = JsonConvert.SerializeObject(pet);
+        var postBody = new StringContent(serializeObject, Encoding.UTF8, mediaType: "application/json");
+        await httpClient.PostAsync(requestUri: "/api/addNewPet", postBody);
+        //when
+        var response = await httpClient.GetAsync(requestUri: "/api/getAllPets");
+        //then
+        response.EnsureSuccessStatusCode();
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```

Comment:
EnsureSuccessStatusCode() has tested the StatuCode， Assert.Equal(HttpStatusCode.OK, response.StatusCode) is not necessary